packages/shared/src/components/onboarding/FilterOnboardingV4.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { ReactElement, useMemo } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import request from 'graphql-request';
+import useFeedSettings from '../../hooks/useFeedSettings';
+import { RequestKey, generateQueryKey } from '../../lib/query';
+import {
+  GET_ONBOARDING_TAGS_QUERY,
+  GET_RECOMMENDED_TAGS_QUERY,
+  Tag,
+  TagsData,
+} from '../../graphql/feedSettings';
+import { graphqlUrl } from '../../lib/config';
+import { disabledRefetch } from '../../lib/func';
+import { Button } from '../buttons/Button';
+import { AlertColor, AlertDot } from '../AlertDot';
+import { SearchField } from '../fields/SearchField';
+import useDebounce from '../../hooks/useDebounce';
+import { useTagSearch } from '../../hooks';
+import type { FilterOnboardingProps } from './FilterOnboarding';
+import useTagAndSource from '../../hooks/useTagAndSource';
+import { Origin } from '../../lib/analytics';
+
+type OnSelectTagProps = {
+  tag: Tag;
+};
+
+const tagsSelector = (data: TagsData) => data?.tags || [];
+
+export function FilterOnboardingV4({
+  className,
+}: FilterOnboardingProps): ReactElement {
+  const queryClient = useQueryClient();
+
+  const { feedSettings } = useFeedSettings();
+  const selectedTags = useMemo(() => {
+    return new Set(feedSettings?.includeTags || []);
+  }, [feedSettings?.includeTags]);
+  const { onFollowTags, onUnfollowTags } = useTagAndSource({
+    origin: Origin.TagsFilter,
+  });
+
+  const [refetchFeed] = useDebounce(() => {
```

Comment:
Just to make sure, you want the debounce to avoid refetching on every tag selection in line?